src/object/srv_obj_migrate.c

```diff
@@ -2009,8 +1963,59 @@ migrate_obj_punch(struct iter_obj_arg *arg)
 	return dss_task_collective(migrate_obj_punch_one, arg, 0);
 }
 
+/* Destroys an object prior to migration. Called exactly once per object ID per
+ * container on the appropriate VOS target xstream.
+ */
+static int
+destroy_existing_obj(struct migrate_pool_tls *tls, unsigned int tgt_idx,
+		     daos_unit_oid_t *oid, uuid_t cont_uuid)
+{
+	struct ds_cont_child *cont;
+	int rc;
+
+	rc = ds_cont_child_open_create(tls->mpt_pool_uuid, cont_uuid, &cont);
```

Comment:
we don't have to open/close container for each object, instead we can keep the open handle and close it until switching to a different container.